specs/directory.md

```diff
@@ -0,0 +1,167 @@
+| `CACHE`   | Directory for user-specific cache                                                   |
+
+Although it is recommended to install `notation` with its plugins and default configurations at the system level, it is possible to install at the user level.
+
+On Unix systems, `notation` follows [Filesystem Hierarchy Standard][FHS] for system level directories and [XDG Base Directory Specification][XDG] for user level directories. On Windows, [Known Folders][KF] and [App Settings][AS] are followed equivalently. On Darwin, [macOS File System][macOS_FS] with [System Integrity Protection][SIP] is followed equivalently. If a file with the same name exists at the system level and the user level, the file at the user level takes over the priority.
+
+### System Level
+
+Default directory paths for various operating systems at system level are specified as below.
+
+| Directory | Unix           | Windows                       | Darwin                         |
+| --------- | -------------- | ----------------------------- | ------------------------------ |
+| `BIN`     | `/usr/bin`     | `%ProgramFiles%/notation/bin` | `/usr/local/bin`               |
+| `LIBEXEC` | `/usr/libexec` | `%ProgramFiles%`              | `/usr/local/lib`               |
+| `CONFIG`  | `/etc`         | `%ProgramData%`               | `/Library/Application Support` |
+
+`CACHE` is omitted since it is user specific.
+
+### User Level
+
+Default directory paths for various operating systems at user level are specified as below.
+
+| Directory | Unix               | Windows          | Darwin                          |
+| --------- | ------------------ | ---------------- | ------------------------------- |
+| `LIBEXEC` | `$XDG_CONFIG_HOME` | `%AppData%`      | `~/Library/Application Support` |
+| `CONFIG`  | `$XDG_CONFIG_HOME` | `%AppData%`      | `~/Library/Application Support` |
+| `CACHE`   | `$XDG_CACHE_HOME`  | `%LocalAppData%` | `~/Library/Caches`              |
+
+There is no default `BIN` path at user level since the `notation` binary can be put anywhere as long as it in the `PATH` environment variable. Common directories on Unix/Darwin are `~/bin` and `~/.local/bin` where manual `PATH` update by users may be required.
+
+## Structure
+
+The overall directory structure for `notation` is summarized as follows.
+
+```
+{BIN}
+└── notation
+{CONFIG}
+└── notation
+    ├── config.json
+    ├── private
+    │   ├── {key-name}.crt
+    │   └── {key-name}.key
+    └── trust
+        ├── policy.json
+        └── store
+            └── {trust-store-type}
+                └── {named-store}
+                    └── {cert-file}
+{CACHE}
+└── notation
+    └── signature
+        └── {manifest-digest-algorithm}
+            └── {manifest-digest}
+                └── {signature-digest-algorithm}
+                    └── {signature-digest}.sig
+{LIBEXEC}
+└── notation
+    └── plugins
+        └── {plugin-name}

```

Comment:
Do you mean Fully qualified name? Two plugin can have same name.

Reply:
Yes, please refer to the plugin spec for details. I will also have an example for that.

Reply:
Added a link to the plugin spec.